hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -285,3 +290,530 @@ func (r *DedicatedServingComponentScheduler) Reconcile(ctx context.Context, req
+			return []reconcile.Request{{NamespacedName: types.NamespacedName{Namespace: namespace, Name: name}}}
+		})).
+		Watches(&schedulingv1alpha1.ClusterSizingConfiguration{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+			hostedClusters := &hyperv1.HostedClusterList{}
+			if err := r.List(ctx, hostedClusters); err != nil {
+				return nil
+			}
+			var out []reconcile.Request
+			for _, hc := range hostedClusters.Items {
+				out = append(out, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: hc.Namespace, Name: hc.Name}})
+			}
+			return out
+		})).
+		Watches(&appsv1.Deployment{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+			deployment := obj.(*appsv1.Deployment)
+			if deployment.Namespace != autoSizerNamespace {
+				return nil
+			}
+			name := deployment.Labels[HostedClusterNameLabel]
+			namespace := deployment.Labels[HostedClusterNamespaceLabel]
+			if name == "" || namespace == "" {
+				return nil
+			}
+			return []reconcile.Request{{NamespacedName: types.NamespacedName{Namespace: namespace, Name: name}}}
+		})).
+		Named(requestServingSchedulerAndSizerName)
+	return builder.Complete(r)
+}
+
+func (r *DedicatedServingComponentSchedulerAndSizer) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	hc := &hyperv1.HostedCluster{}
+	log := ctrl.LoggerFrom(ctx)
+	err := r.Get(ctx, req.NamespacedName, hc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("hostedcluster not found, aborting reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
+	}
+	if !hc.DeletionTimestamp.IsZero() {
+		log.Info("hostedcluster is deleted, nothing to do")
+		return ctrl.Result{}, nil
+	}
+	if hcTopology := hc.Annotations[hyperv1.TopologyAnnotation]; hcTopology != hyperv1.DedicatedRequestServingComponentsTopology {
+		log.Info("hostedcluster does not use isolated request serving components, nothing to do")
+		return ctrl.Result{}, nil
+	}
+	isPaused, duration, err := util.ProcessPausedUntilField(hc.Spec.PausedUntil, time.Now())
+	if err != nil {
+		log.Error(err, "error processing hosted cluster paused field")
+		return ctrl.Result{}, nil // user needs to reformat the field, returning error is useless
+	}
+	if isPaused {
+		log.Info("Reconciliation paused", "pausedUntil", *hc.Spec.PausedUntil)
+		return ctrl.Result{RequeueAfter: duration}, nil
+	}
+
+	config := schedulingv1alpha1.ClusterSizingConfiguration{}
+	if err := r.Get(ctx, types.NamespacedName{Name: "cluster"}, &config); err != nil {
+		return ctrl.Result{}, fmt.Errorf("could not get cluster sizing configuration: %w", err)
+	}
+
+	configValid := false
+	for _, condition := range config.Status.Conditions {
+		if condition.Type == schedulingv1alpha1.ClusterSizingConfigurationValidType && condition.Status == metav1.ConditionTrue {
+			configValid = true
```

Comment:
I think we should break here once the condition is found even if the status is not true.
how about using 
```golang
meta.FindStatusCondition(config.Status.Conditions, schedulingv1alpha1.ClusterSizingConfigurationValidType)
```